OSDOCS:2475: Adding Dynamic Plugins as Tech Preview in Rel Notes (#40759)
openshift/openshift-docs: release_notes/ocp-4-10-release-notes.adoc

[id="ocp-4-10-web-console"]
=== Web console

[id="ocp-4-10-dynamic-plugin-technology-preview"]
==== Dynamic Plug-in (Technology Preview)

> Starting with {product-title} 4.10, OpenShift console dynamic plug-ins are now available as a Technology Preview feature. This feature allows you make customizations to your interface at runtime such as: add custom pages, add perspectives and update navigation items, and add tabs and actions to resource pages. Plug-ins are registered with the console using the `ConsolePlugin` custom resource, and they are enabled in the console operator configuration by a cluster administrator.

Author: See #41260 for the docs.

[id="ocp-4-10-multicluster-console-technology-preview"]
==== Multi-cluster Console (Technology Preview)

> Starting with {product-title} 4.10, the multi-cluster console is now available as a Technology Preview feature. This feature supports a single {product-title} console that can connect to remote clusters API. An *All clusters* perspective contains the Advanced Cluster Management(ACM) multi-cluster views. This feature also implements single sign-on(SSO), so you can bypass the login prompt when navigating between clusters.

Reviewer: We'll need to document how to enable the multi-cluster feature somewhere. It requires setting a feature gate.

The "All clusters" perspective will only be available if you have ACM 2.5 installed and enable the ACM console plugin, which is a manual step an admin needs to take.

I'd remove or rephrase the sentence about SSO. We have an IDP config operator that will push configuration for either GitHub or LDAP identity providers to all clusters in the fleet, but it's a separate operator to install and configure.

cc @alimobrem @showeimer @jamestalton @robinbobbitt

[id="ocp-4-10-web-console"]
=== Web console

[id="ocp-4-10-dynamic-plugin-technology-preview"]
==== Dynamic plug-in (Technology Preview)

> Starting with {product-title} 4.10, OpenShift console dynamic plug-ins are now available as a Technology Preview feature. This feature allows you make customizations to your interface at runtime such as: add custom pages, add perspectives and update navigation items, and add tabs and actions to resource pages. Plug-ins are registered with the console using the `ConsolePlugin` custom resource, and they are enabled in the console operator configuration by a cluster administrator.

Reviewer 2: I might add a list here and try to make the first part more user-focused. (Saying that the program allows you to do something is a bit anthropomorphic.) Maybe something like:

```suggestion
Starting with {product-title} {product-version}, the ability to create OpenShift console dynamic plug-ins is now available as a Technology Preview feature. You can use this feature to customize your interface at runtime in many ways, including:

* Adding custom pages
* Adding perspectives and updating navigation items
* Adding tabs and actions to resource pages

After you register a plug-in with the console by using the `ConsolePlugin` custom resource, a cluster administrator must enable them in the console Operator configuration.
```

Are you creating multiple plug-ins? (I want to make sure we're describing the single feature correctly. It sounds like the feature is the ability to make a plug-in.)
Can anyone register the plug-in?

Author: You actually don't need admin privileges. Updating this sentence.

Reviewer: You need to have authority to create a `ClusterPlugin` resource. By default, only cluster admins can create one.

For more information on the dynamic plug-in, see xref:../web_console/dynamic-plug-ins.adoc#dynamic-plug-ins_dynamic-plug-ins[Adding a dynamic plug-in to the {product-title} web console].

[id="ocp-4-10-multicluster-console-technology-preview"]
==== Multi-cluster console (Technology Preview)

> Starting with {product-title} 4.10, the multi-cluster console is now available as a Technology Preview feature. This feature supports a single {product-title} console that can connect to remote clusters API.

Reviewer 2: Maybe:

```suggestion
Starting with {product-title} 4.10, the multi-cluster console is now available as a Technology Preview feature. By enabling this feature, you can connect to remote clusters' API servers from a single {product-title} console.
```

Author: Added suggestion.

|-
|TP

> |Pod-level bonding for secondary networks

Author: I think something happened in rebase here.
